sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -155,6 +157,15 @@ final class DataFrameWriter private[sql](df: DataFrame) {
     this.sortColumnNames = Option(colName +: colNames)
     this
   }
+  /*
+   * Specify the compression codec when saving it on hdfs.
+   *
+   * @since 2.0.0
+   */
+  def compress(codec: Class[_ <: CompressionCodec]): DataFrameWriter = {
```

Comment:
Also we shouldn't depend on Hadoop APIs in options, which is a user facing API. Nobody outside the Hadoop world knows how to use the CompressionCodec API.